src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,18 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {

```

Comment:
The class `ExcavatorProducer` is correctly parameterized with the `Excavator` type, which is a subtype of `Machine`, ensuring that non-machine types cannot be used here.